src/store.ts

```diff
@@ -453,6 +458,83 @@ export default function Store(
 
         mpInstance._Persistence.savePersistence(this.persistenceData);
     };
+
+    this.getFirstSeenTime = (mpid: MPID) => {
+        if (!mpid) {
+            return null;
+        }
+
+        if (
+            this.persistenceData &&
+            this.persistenceData[mpid] &&
+            this.persistenceData[mpid].fst
+        ) {
+            return this.persistenceData[mpid].fst;
+        } else {
+            return null;
+        }
+    };
+
+    this.setFirstSeenTime = (mpid: MPID, _time?: number) => {
+        if (!mpid) {
+            return;
+        }
+
+        const time = _time || new Date().getTime();
+
+        if (this.persistenceData) {
```

Comment:
Do we ever have to check if `this.persistenceData` doesn't exist if there will always be persisetenceData [as part of the default store creation](https://github.com/mParticle/mparticle-web-sdk/pull/855/files#diff-2717c7fb31c2070dd96f07394997809bc9be7f3fff03ab638db696cc480f39b6R225)?

Reply:
i'm not opposed to playing it safe.  i thought TS might help us here though.

Reply:
As mentioned above, I'm copying over the existing functions as is (or close to as is) without optimization to keep the changes as small as possible. I think there's a few places where we can optimize or DRY things up after we port overt he functionality, but that's out of scope IMO.